### ISSUE TITLE
Disable tooltips with empty content by default

### ISSUE THREAD

**Issue by adidahiya** [palantir/blueprint#886]
There's no reason to show empty tooltips. I would prefer if Blueprint prevented me from shooting myself in the foot here:

![image](https://cloud.githubusercontent.com/assets/723999/24312016/11fea9f4-10ad-11e7-933f-3dcf6d1e9d4c.png)

If you agree @giladgray @llorca, I can send a PR

**Comment by giladgray** [comment]
How did you do that? `content` is required.

**Comment by cmslewis** [comment]
Makes sense. Worth printing a warning to the console to explain why the tooltip isn't showing up, when this happens? Could also add an explanatory sentence to a prop description. I prefer the second approach I guess.

On Mar 24, 2017, at 1:16 PM, Adi Dahiya <notifications@github.com<mailto:notifications@github.com>> wrote:


There's no reason to show empty tooltips. I would prefer if Blueprint prevented me from shooting myself in the foot here:

[image][cloud.githubusercontent.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__cloud.githubusercontent.com_assets_723999_24312016_11fea9f4-2D10ad-2D11e7-2D933f-2D3dcf6d1e9d4c.png&d=DwMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=-xHM_msPNZ5Ueek6I-8Uf3C2NKIdRgHQEGOCuANP978&m=8hpSq1qXl1l-RivXhUGLlcP6rRxDEaK6trL2CbX-_iE&s=K7xP8NxFlmjywxPJQ5Ag0ZsBtUlKh6pv35WSppAEXbM&e=>

If you agree @giladgray[github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_giladgray&d=DwMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=-xHM_msPNZ5Ueek6I-8Uf3C2NKIdRgHQEGOCuANP978&m=8hpSq1qXl1l-RivXhUGLlcP6rRxDEaK6trL2CbX-_iE&s=bco-9MCno4jrEQt6zllPwsQmL1mG8PfMLkhH-Sc3MhI&e=> @llorca[github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_llorca&d=DwMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=-xHM_msPNZ5Ueek6I-8Uf3C2NKIdRgHQEGOCuANP978&m=8hpSq1qXl1l-RivXhUGLlcP6rRxDEaK6trL2CbX-_iE&s=b0vftA9tmrTVHWUdJI-pyTwA01_aTBCvCt_-Luas-vI&e=>, I can send a PR

**Comment by adidahiya** [comment]
@giladgray pretty simple:

```ts
content={null}
content=""
content={undefined}
```

**Comment by leebyp** [comment]
>Worth printing a warning to the console to explain why the tooltip isn't showing up, when this happens? Could also add an explanatory sentence to a prop description.

how about... both?

**Comment by adidahiya** [comment]
> Worth printing a warning to the console to explain why the tooltip isn't showing up

No, please don't do that. Sometimes I have dynamic data and can't guarantee that it will be non-empty. The whole point of asking for this feature is so that I don't have to do the check for empty contents. Of course I could implement this on my own right now but I prefer Blueprint to handle it:

```ts
<Tooltip
    disabled={content == null || content === ""}
    content={content}
>
```

**Comment by giladgray** [comment]
i'm not going to handle the whitespace string `"    "` case cuz that's not technically empty.

**Comment by giladgray** [comment]
also doing this only for `Tooltip`, not `Popover`.

**Comment by alisabzevari** [comment]
Shouldn't you also check if the tooltip `isOpen` is true? It keeps showing the warning even if `isOpen` is false.

**Comment by adidahiya** [comment]
@alisabzevari that was fixed in 1.15.0 https://github.com/palantir/blueprint/issues/1000
